api/v1alpha6/types.go

```diff
@@ -145,6 +145,9 @@ type PortOpts struct {
 	// These tags are applied in addition to the instance's tags, which will also be applied to the port.
 	// +listType=set
 	Tags []string `json:"tags,omitempty"`
+
+	// Value specs are extra parameters to include in the request.
+	ValueSpecs *map[string]string `json:"valueSpecs,omitempty"`
```

Comment:
I know we use them elsewhere (just above in Profile, for eg), and CAPI uses them extensively, but use of maps like this [goes against k8s API guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps). We at least shouldn't add more of them.

Please could you replace this with a list with a name as a key element?

I also wonder if we could improve the docstring.